### PR TITLE
Feature 415 post request

### PR DESCRIPTION
**Pull request by gustavo-rojas-jalafund**
AB# 415 Frontend | Implement Post request to create an postulation

https://dev.azure.com/funjaladev/DevLevel3/_workitems/edit/415

task (3)

### CODE REVIEW

File: src/components/projects/ModalProjects.jsx [hunk 1]
```diff
@@ -47,13 +47,84 @@ export default function ModalProjects({ idResume, setModalProjects }) {
...
+        projectId: project.id,
+        projectName: project.name,
+        resumeId: idResume,
+        resumeName: "prueba",
+        state: "Applied",
+      };
+      const response = await postPostulation(postulation);
+    }
+
+    // onClose();
```
Comment (ela-ocana-jalafund) [inline src/components/projects/ModalProjects.jsx]: Delete this comment :)

File: src/components/projects/ModalProjects.jsx [hunk 2]
```diff
@@ -47,13 +47,84 @@ export default function ModalProjects({ idResume, setModalProjects }) {
...
+      creationDate: "2021-02-05",
+      id: 1,
+      lastUpdate: "2021-11-03",
+      picture: "photo.png",
+      projectId: "724d912-59de-44bd-a092-f5d8434584de",
+      projectName: "Team Up",
+      resumeId: "12",
+      resumeName: "Jose Ecos",
+      state: "Applied",
+    });
```
Comment (paola-quintanilla-jalafund) [inline src/components/projects/ModalProjects.jsx]: delete this project push

File: src/components/projects/ModalProjects.jsx (hunk 1 above)
Comment (paola-quintanilla-jalafund) [inline src/components/projects/ModalProjects.jsx]: delete comment

File: src/components/projects/ModalProjects.jsx (hunk 2 above)
Comment (gustavo-rojas-jalafund) [reply src/components/projects/ModalProjects.jsx]: Done

File: src/components/projects/ModalProjects.jsx (hunk 1 above)
Comment (gustavo-rojas-jalafund) [reply src/components/projects/ModalProjects.jsx]: Done

File: src/components/projects/ProjectsAPI.js [hunk 3]
```diff
@@ -80,3 +80,24 @@ export async function getProjectBySkill(skill) {
     return;
   }
 }
+
+export async function postPostulation(postulation) {
+  try {
+    const { data } = await axios.post(
+      `${API_HOST}api/${API_VERSION}/postulation`,
```
Comment (ela-ocana-jalafund) [inline src/components/projects/ProjectsAPI.js:87]: Should be "postulations" with s at the end

File: src/components/projects/ProjectsAPI.js [hunk 4]
```diff
@@ -85,6 +85,7 @@ export async function postPostulation(postulation) {
   try {
     const { data } = await axios.post(
       `${API_HOST}api/${API_VERSION}/postulation`,
+      // "http://localhost:7071/api/v1/postulations",
```
Comment (ela-ocana-jalafund) [inline src/components/projects/ProjectsAPI.js]: could you delete this comment please. :)

File: src/components/projects/ProjectsAPI.js [hunk 5]
```diff
@@ -80,3 +80,25 @@ export async function getProjectBySkill(skill) {
     return;
   }
 }
+
+export async function postPostulation(postulation) {
+  try {
+    const { data } = await axios.post(
+      `${API_HOST}api/${API_VERSION}/postulation`,
+      // "http://localhost:7071/api/v1/postulations",
```
Comment (jorge-lema-jalafund) [inline src/components/projects/ProjectsAPI.js]: remove comment

**Review by jorge-lema-jalafund — APPROVED** [review]
approved

**Review by jorge-lema-jalafund — CHANGES_REQUESTED** [review]
make changes

File: src/components/projects/ProjectsAPI.js (hunk 5 above)
Comment (gustavo-rojas-jalafund) [reply src/components/projects/ProjectsAPI.js]: Done

**Review by paulo-teran-jalafund — DISMISSED** [review]

**Review by lilian-delgado-jalafund — DISMISSED** [review]

File: src/components/projects/ModalProjects.jsx [hunk 6]
```diff
@@ -47,13 +47,70 @@ export default function ModalProjects({ idResume, setModalProjects }) {
...
+
+    for (const resume of postulations) {
+      const postulation = {
+        creationDate: new Date().toDateString(),
+        lastDate: date,
+        picture: project.logo,
+        projectId: project.id,
+        projectName: project.name,
+        resumeId: idResume,
+        resumeName: "prueba",
```
Comment (jorge-lema-jalafund) [inline src/components/projects/ModalProjects.jsx]: why hardcoded value

File: src/components/projects/ModalProjects.jsx [hunk 7]
```diff
@@ -47,13 +47,70 @@ export default function ModalProjects({ idResume, setModalProjects }) {
...
+
+    for (const project of postulationsSelected) {
+      const postulation = {
+        creationDate: new Date().toDateString(),
+        lastDate: date,
+        picture: project.logo,
+        projectId: project.id,
+        projectName: project.name,
+        resumeId: idResume,
+        resumeName: "prueba",
```
Comment (jorge-lema-jalafund) [inline src/components/projects/ModalProjects.jsx]: why hardcoded value

**Review by jorge-lema-jalafund — CHANGES_REQUESTED** [review]
approved

File: src/components/projects/ModalProjects.jsx (hunk 1 above)
Comment (gustavo-rojas-jalafund) [reply src/components/projects/ModalProjects.jsx]: Done

Comment (gustavo-rojas-jalafund) [reply src/components/projects/ModalProjects.jsx]: Done

File: src/components/projects/ProjectsAPI.js (hunk 4 above)
Comment (gustavo-rojas-jalafund) [reply src/components/projects/ProjectsAPI.js]: Done

File: src/components/projects/ModalProjects.jsx (hunk 7 above)
Comment (gustavo-rojas-jalafund) [reply src/components/projects/ModalProjects.jsx]: I change for project.resumeName

**Review by lilian-delgado-jalafund — APPROVED** [review]

File: src/components/projects/ModalProjects.jsx (hunk 6 above)
Comment (gustavo-rojas-jalafund) [reply src/components/projects/ModalProjects.jsx]: I change for project.resumeName

**Review by paulo-teran-jalafund — APPROVED** [review]

**Review by jorge-lema-jalafund — APPROVED** [review]

**Pull request merged**